### ISSUE TITLE
procmail on 18.04 (in reality  a netplan issue)

### ISSUE THREAD

**Issue by ahyear** [virtualmin/virtualmin-install#13]
hello,

After a fresh install, when i recheck configuration i get : 
The Procmail program needed for spam filtering does not appear to be installed on your system, or has not yet been set up properly in Webmin's Procmail Mail Filter module. If your system does not use spam filtering, it should be disabled in Virtualmin's module configuration page.

EDIT : 
fixed by adding mailbox_command = /usr/bin/procmail to /etc/postfix/main.cf

**Comment by jcameron** [comment]
Did you also need to run `apt-get install procmail` ?

**Comment by ahyear** [comment]
Yes it was already installed but i added the line on my edit part and it fixed the error (or so it seams)

**Comment by swelljoe** [comment]
You don't want procmail in mailbox_command. You want procmail-wrapper. Was that what was there before and you changed it or was it not set at all? (That determines what I need to do to fix the problem.)

**Comment by swelljoe** [comment]
Also, procmail-wrapper needs to be installed, too. It should be in our repo, but maybe I made a mistake somewhere.

**Comment by ahyear** [comment]
Proclaim was installed but There was no mailbox_command on the /etc/postfix/main.cf

**Comment by swelljoe** [comment]
I'm not asking about procmail. I'm asking about `procmail-wrapper` which is a different package and a different command.

I just did a test install on Ubuntu 18.04 and the mailbox_command got set correctly and the procmail-wrapper command did get installed...so, something must have gone wrong earlier in your installation that prevented the Postfix configuration from getting done. Were there errors during installation, particularly during stage 3 (Configuration)? You can look in the virtualmin-install.log for clues.

Someone else ran into network configuration failure on 18.04 because they've moved to using netplan instead of the old networking scripts found on earlier Ubuntu versions...I didn't know that, as the VMs I was using for testing at Vultr had already been configured to use the old way and so my installs went off without a hitch.

So...did the Network config step succeed? Do you have an /etc/network/interfaces file on your system? I'm guessing there was a failure somewhere before Postfix and Procmail (and Network comes before those)...it should have provided an error, if so.

**Comment by swelljoe** [comment]
So, for now, we need the /etc/network/interfaces file to exist *and* be the actual network configuration for the system; you need to remove the netplan configuration file for networking and add the old style config. It has to be the actual network configuration of the system; you can't just create an empty file there (or a file with bogus data); we're reading it because we need data out of it, not for fun. ;-)

Jamie will be working on supporting the new netplan thing...but, that'll take a while. Probably a couple of weeks, at least. Jamie works fast, but this is a whole new thing with its own configuration files and tools. It's not just moving old files to new locations, it's supporting an entirely new network configuration system.

If you do a fresh install, set it up to use /etc/network instead of netpland/etworkd. How to do that...is kinda beyond my knowledge. From what I can tell, you just have to remove the networkd/netplan configuration, and add the right files into /etc/network.

I don't plan to change the install script to do this for you, as it would be pretty risky for us to poke at network configuration in such a dramatic way. And, since Jamie is adding support to Webmin for this new thing, we can just wait a bit and I'll detect it in the future and configure whichever network system is in use when the installation runs.

The only thing I'd add about a fresh install is *watch the install* looking for errors. You pretty much can't ignore errors, or just work around the one error you see and then expect it to be good. If the installer fails, it really failed, and we need to work on the very first thing that failed and fix that and then do another install, until all errors are resolved.

There have been several discussions on the Ask Ubuntu site about this change, and some discussion of how to revert to the old way; it doesn't *seem* complicated, but I haven't done it yet, so I can't give you the steps with any confidence.

**Comment by ahyear** [comment]
Ok. it's cristal clear.
as I'm not a ubuntu/linux rocket scientist ( you don't say by the look of my solutions... ^^ ) i'd rather wait for Jamie to do magic before dooing a fresh install. 
and if I change to the old network configuration it's not said that it won't bite me in the ass in a few months... ^^
so if it's ok with you, I'll wait for an update of the netplan integration.

Thank you

**Comment by swelljoe** [comment]
I think that's a perfectly reasonable plan. You could also deploy on 16.04 or Debian 9, where we don't see this issue, for now, and migrate to 18.04 in a couple of weeks when we get this sorted.

**Comment by ahyear** [comment]
Hello,

Do you have any news on the topic?

Thank you again for your work...

**Comment by swelljoe** [comment]
Still waiting for netplan support in Webmin. I'll check in with Jamie today to see if he's had a chance to work on it yet.

**Comment by swelljoe** [comment]
I also found a couple days ago there's also an issue even with the old way of configuring network interfaces, where BIND doesn't get setup correctly, either, so I have some troubleshooting to do today, as well.

**Comment by jcameron** [comment]
I've started working on Ubuntu 18 networking support, but it's slow going because Ubuntu 18 changed a lot of things :-(

**Comment by ahyear** [comment]
well at least I helped find an issue on Ubuntu 18 installation ^^.
should I open my issue on virtualmin-gpl then? 
because it's more a virtualmin issue than a installer issue...

**Comment by paulocoghi** [comment]
@ahyear, since the issues with 18.04 are known, I think you don't need to open another issue there.

**Comment by swelljoe** [comment]
Yeah, we're on it. It's not a Virtualmin issue...it's a virtualmin-install and Webmin issue and both me and Jamie know about it (I'm waiting on Jamie to roll a new Webmin with the netplan support, and then I'll do another virtualmin-install and/or virtualmin-config release with any needed changes to make it work).

**Comment by ahyear** [comment]
for info, I used this tutorial and got no error:
https://www.youtube.com/watch?v=s_iWXrsvOHQ

(I just did not do the allow part in miniserv.conf)

**Comment by scothlz** [comment]
Today, Clean Ubuntu 18.04 on netplan working ipv4 and ipv6, webmin installed first.  Then Virtualmin install via install.sh.  Still get this 2 postfix errors upon "checking configuration".  
webmin > servers > postfix > virtual domains > "Domain mapping lookup tables"  was blank.  Added hash:/etc/postfix/virtual.
Then it failed with error of this original thread starter.  Fixed by adding mailbox_command = /usr/bin/procmail-wrapper to /etc/postfix/main.cf.    NOTE:  original solution gave me permissions error on /usr/bin/procmail as its owned by root:mail.  Where as /usr/bin/procmal-wrapper is owned by root:root and worked.  Just thought I'd offer solution all in one place as I run into this on almost every install going back quite a while.

**Comment by paulocoghi** [comment]
@swelljoe , I installed on Ubuntu 18.04 with zero issues. Everything is working perfectly. I believe this issue can be closed.

**Comment by paulocoghi** [comment]
Just one issue, for now, documented on https://github.com/virtualmin/virtualmin-install/issues/22

**Comment by paulocoghi** [comment]
Also, DKIM service isn't created by Virtualmin (6.06) on Ubuntu 18.04. Details [here](https://github.com/virtualmin/virtualmin-install/issues/25).

**Comment by swelljoe** [comment]
Pretty sure these netplan related issues are resolved. We haven't had new comments about it in a while, and there have been a couple of Virtualmin/Webmin releases.

**Comment by voarsh2** [comment]
I got this recently after installing from .sh file

**Comment by iliarostovtsev** [comment]
@voarsh2 

> I got this recently after installing from .sh file

What an exact problem that you got?

**Comment by voarsh2** [comment]
> What an exact problem that you got?

I can only run a ngrok terminal session with a wildcard.
I cannot use the YAML to put a wildard domain.
